### PR TITLE
Empty weight

### LINKED ISSUE #46: Remove "No Weight" and "Too Small" buttons
>interferes with smoothness of data entry and adds no value.
>We will use NULL value when weight field is not populated, or whatever placeholder exists for entries without weight (appears to be -1) and those too small to be measured 
(appears to be 0).

Ref 6) in https://docs.google.com/document/d/1pTsCfe0-Kmp20vSLy9DJtVIEFB_PwfoabOojmtVQXYM/edit

### LINKED ISSUE #47: Allow negative values for Length field
Remove any validation that prevents negative number.

Ref 7) in https://docs.google.com/document/d/1pTsCfe0-Kmp20vSLy9DJtVIEFB_PwfoabOojmtVQXYM/edit

### PR DESCRIPTION
**Pull request by steveoh**
This seems to work pretty well. Looking forward to the review. 

closes #46 and closes #47

### CODE REVIEW

**Comment by steveoh** [comment]
This needs to allow for decimals now...

**Comment by steveoh** [comment]
decimals were supported. this is ready for a looksy

File: src/app/location/Location.js [hunk 1]
```diff
@@ -246,7 +246,9 @@ define([
                     // this needs to be here since it sets the text on the verify btn
                     // and it needs to wait for $(that.verifyMapBtn).button('loading'); to finish
                     // which is async
-                    that.addLineToMap(inProgressData.path);
+                    if (inProgressData && inProgressData['path']) { // eslint-disable-line dot-notation
```
Comment (stdavis) [inline src/app/location/Location.js]: No need to used []'s here. Missing property will just return undefined.
image

Comment (steveoh) [reply src/app/location/Location.js]: The issue is that in progress data was null which does throw.

Comment (stdavis) [reply src/app/location/Location.js]: Yes, I understand that. I was just suggesting something like this:
```js
if (inProgressData && inProgressData.path) {
...
```

No need for brackets or eslint stuff...

**Pull request merged**